compiler/rustc_parse/src/parser/item.rs

```diff
@@ -2719,6 +2721,17 @@ impl<'a> Parser<'a> {
                                 ).note("keyword order for functions declaration is `pub`, `default`, `const`, `async`, `unsafe`, `extern`");
                         }
                     }
+                    // The user wrote incompatible keywords like `safe unsafe` or `unsafe safe`
+                    else if let Some(WrongKw::Incompatible { first, second }) = wrong_kw {
+                        if let Ok(first_kw) = self.span_to_snippet(first)
+                            && let Ok(second_kw) = self.span_to_snippet(second)
+                        {
```

Comment:
Remark: this is more of a band-aid fix, because this recovery code path is taken when we just so happen to be trying to parse anything that's fn-frontmatter-like that does not immediately include an `fn` keyword (as part of trying to parse item shaped things). So it's kinda best-effort anyway.

Reply:
safe isnt supposed to be incompatible with safe in this context
```
safe unsafe extern {} // safe is only valid inside of this block
                                 // Should give the err: safe is only valid inside of an unsafe extern block
```
so this error should be for the extern block
```
unsafe extern {
    unsafe safe fn foo() {} // Err: safe and unsafe are incompatible
}
```
this is atleast what i understand
here is a valid use of safe:
```
unsafe extern {
    safe fn foo() -> i32{
        42
    }
}
// Now foo can be called safely
println!("{}", foo()); // out: 42
```